### ISSUE TITLE
`Illegal header value b'Bearer '` with Non-OpenAI LLMs

### ISSUE THREAD

**Issue by gaspardbb** [langroid/langroid#308]
I would be interested in trying out langroid but don't want to go through OpenAI API for prototyping. Unfortunately, I find setting up ollama not straightforward [in the tutorial](https://langroid.github.io/langroid/tutorials/non-openai-llms).
1. Using LiteLLM as a proxy server does not work directly for me (`chat_model="local/localhost:8000"`)
2. [Using LiteLLM directly](https://langroid.github.io/langroid/tutorials/non-openai-llms/#using-the-litellm-library) does work (`chat_model="litellm/ollama/mistral"`)

This bug report concerns option 1, which is described first in your tutorial. 

## Describe the bug
When using langroid with ollama and [liteLLM as shown in the tutorial](https://langroid.github.io/langroid/tutorials/non-openai-llms/#litellm-openai-proxy-server), the `llm.chat` method loops with timeout error. 
When debugging, the exception is 
```
Traceback (most recent call last):
  File "<python_lib>/httpcore/_exceptions.py", line 10, in map_exceptions
    yield
  File "<python_lib>/httpcore/_sync/http11.py", line 142, in _send_request_headers
    event = h11.Request(
            ^^^^^^^^^^^^
  File "<python_lib>/h11/_events.py", line 96, in __init__
    self, "headers", normalize_and_validate(headers, _parsed=_parsed)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<python_lib>/h11/_headers.py", line 164, in normalize_and_validate
    validate(_field_value_re, value, "Illegal header value {!r}", value)
  File "<python_lib>/h11/_util.py", line 91, in validate
    raise LocalProtocolError(msg)
h11._util.LocalProtocolError: Illegal header value b'Bearer '
```

## To Reproduce

With litellm running, e.g. `litellm --model ollama/mistral`, 

```python
from langroid.language_models.openai_gpt import OpenAIGPTConfig, OpenAIGPT
from langroid.language_models.base import LLMMessage, Role

# create the (Pydantic-derived) config class: Allows setting params via MYLLM_XXX env vars
MyLLMConfig = OpenAIGPTConfig.create(prefix="myllm") 

# instantiate the class, with the model name and context length
my_llm_config = MyLLMConfig(
    # chat_model="litellm/ollama/mistral",  # This works 
    chat_model="local/localhost:8000",
    chat_context_length=2048,  # adjust based on model
)

llm = OpenAIGPT(my_llm_config)
messages = [
    LLMMessage(content="You are a helpful assistant",  role=Role.SYSTEM),
    LLMMessage(content="What is the capital of Ontario?",  role=Role.USER),
]
response = llm.chat(messages, max_tokens=50)
```

## Version 

`langroid  0.1.125`

Happy to provide more details if needed, I'm not familiar with openai APIs and maybe I overlooked sth obvious.

**Comment by pchalasani** [comment]
I just tried this with `litellm --model ollama/llama2`, and it shows the specific URL where it is listening:
```
INFO:     Uvicorn running on http://0.0.0.0:7912 (Press CTRL+C to quit)
```
Then in the code above, I use `chat_model = "local/localhost:7912"` and it works fine. 

I think your error is that you did not use the right port number. 
Let me know if this doesn't resolve the issue for you.

**Comment by gaspardbb** [comment]
Unfortunately this is something I checked, litellm is running at the correct port number. 

Those are some attributes of the POST request in the `Request` object from `httpx/_client.py:1002`, which seem correct:
```
method: 'POST',
url: URL('http://localhost:8000/chat/completions'),
content: b'{"messages": [{"role": "system", "content": "You are a helpful assistant"}, {"role": "user", "content": "What is the capital of Ontario?"}], "model": "local/localhost:8000", "max_tokens": 50, "n": 1, "stop": null, "stream": true, "temperature": 0.2}'
```

**Comment by pchalasani** [comment]
Puzzling. The other thing I can think of is to check the `litellm` version -- I am on `1.3.3`

**Comment by gaspardbb** [comment]
Hm, indeed I was on `litellm v1.0.3`. But upgrading to `1.3.3` produces the same error (making sure litellm is running). 
I'll try to debug this more if I have more!

**Comment by pchalasani** [comment]
@gaspardbb I think I know what’s going on. The OpenAI API Key needs to be non empty, when using a local model via the proxy. We fixed it in the code so it is set to “xxx” when it’s absent in the environment. Please try upgrading to 0.1.128 and let us know if it’s not fixed

**Comment by gaspardbb** [comment]
Indeed, the issue is fixed with `0.1.128`. Nice catch! I'm closing the issue.
